modules/native/native-grammar.ym

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2025 BalaBit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+%code top {
+#include "native-parser.h"
+
+}
+
+
+%code {
+
+#include "cfg-parser.h"
+#include "parser.h"
+#include "native-grammar.h"
+#include "syslog-names.h"
+#include "messages.h"
+#include "plugin.h"
+#include "cfg-grammar.h"
+
+#include <string.h>
+
+__attribute__((visibility("hidden"))) LogParser*
+native_parser_new(GlobalConfig *cfg);
+
+}
+
+%name-prefix "native_"
+
+/* this parameter is needed in order to instruct bison to use a complete
+ * argument list for yylex/yyerror */
+
+%lex-param {CfgLexer *lexer}
+%parse-param {CfgLexer *lexer}
+%parse-param {LogParser **instance}
+%parse-param {gpointer arg}
+
+/* INCLUDE_DECLS */
+
+%token KW_OPTION
+
+%type <ptr> native
+%type <ptr> native_parser
+%type <ptr> native_parser_params
+%%
+
+start
+  : native
+    {
+      *instance = $1;
+      if (yychar != YYEMPTY)
+        cfg_lexer_unput_token(lexer, &yylval);
+      YYACCEPT;
+    }
+  ;
+
+native
+  : LL_CONTEXT_PARSER native_parser { $$ = $2; }
+  ;
+
+native_parser
+  : LL_IDENTIFIER '(' native_parser_params ')' { $$ = *instance; }
+  ;
+
+native_parser_params
+  : {
+      *instance = native_parser_new(configuration);
+
+      if (!*instance)
+       {
+         msg_error("Cannot create the Native parser, aborting",
+                   NULL);
+         YYERROR;
+       }
+    }
+    native_parser_options
+  ;
+
+native_parser_options
+  : native_parser_option native_parser_options
+  |
+  ;
+
+native_parser_option
+  : KW_OPTION '(' string string ')'
+    {
+      native_parser_set_option(*instance, $3, $4);
```

Comment:
I think one of the reason to use this binding it to simplify writing modules. If the rust grammar is directly hooked into C I think I lose this simplicity. We can wrap the modules in SCLs which can hide the configuration ugliness. I think a way to express a 'verbatim' text and passing it to a rust grammar would be better, but maybe the two approaches can live side by side. First I'd use the SCL and option based one and then we may come up with something smarter.

The `LogParser` struct could be filled by rust but I wanted to keep `LogParser` as an opaque pointer. I tried to minimize the size of the C side and forwarding a function call is the simplest thing I can come up with.